compiler/qsc/src/interpret/stateful.rs

```diff
@@ -60,49 +61,19 @@ impl Error {
 enum ErrorKind {
     #[error(transparent)]
     #[diagnostic(transparent)]
-    Compile(#[from] compile::Error),
+    Compile(#[from] WithSource<compile::Error>),
+    #[error(transparent)]
+    #[diagnostic(transparent)]
+    Incremental(#[from] WithSource<incremental::Error>),
     #[error(transparent)]
     #[diagnostic(transparent)]
-    Pass(#[from] qsc_passes::Error),
+    Pass(#[from] WithSource<qsc_passes::Error>),
     #[error("runtime error")]
     #[diagnostic(transparent)]
-    Eval(#[from] WithStack<qsc_eval::Error>),
+    Eval(#[from] WithStack<WithSource<qsc_eval::Error>>),
     #[error("entry point not found")]
     #[diagnostic(code("Qsc.Interpret.NoEntryPoint"))]
     NoEntryPoint,
-}
-
-#[derive(Clone, Debug, Diagnostic, Error)]
-#[diagnostic(transparent)]
-#[error(transparent)]
-pub struct LineError(WithSource<LineErrorKind>);
```

Comment:
This is just a cosmetic refactoring: `LineError` and `Error` were extremely similar enums that could be combined into one enum. I find this a bit easier to read.